OrbitCore/ElfFile.cpp

```diff
@@ -220,8 +220,19 @@ bool ElfFileImpl<llvm::object::ELF32LE>::Is64Bit() const {
 
 }  // namespace
 
+std::unique_ptr<ElfFile> ElfFile::Create(const char *buf, size_t len) {
+  std::unique_ptr<llvm::MemoryBuffer> buffer = llvm::MemoryBuffer::getMemBuffer(llvm::StringRef(buf, len), llvm::StringRef("buffer name"), false);
+  llvm::Expected<std::unique_ptr<llvm::object::ObjectFile>> object_file_or_error = llvm::object::ObjectFile::createObjectFile(buffer->getMemBufferRef());
+
+  if (!object_file_or_error) {
+    return nullptr;
+  }
+
+  return ElfFile::Create("INMEMORY", llvm::object::OwningBinary<llvm::object::ObjectFile>(std::move(object_file_or_error.get()), std::move(buffer)));
```

Comment:
if this method is used only for testing this is fine - otherwise I would have added file_name or elf_name parameter to Create method...

Reply:
Good point. I will add the parameter.

Reply:
I added a file_path parameter to the new Create function. But I felt this function should no longer belong to the same overload-set as the other Create functions. Since it is not reading from the file this would be highly misleading, so I renamed it to CreateFromBuffer.

At the same time, I cleaned up the overload set by replacing std::string by std::string_view.

Reply:
I added a test for `CreateFromBuffer`:
```cpp
TEST(ElfFile, CreateFromBuffer) {
  std::string executable_path = Path::GetExecutablePath();
  std::string test_elf_file = executable_path + "/testdata/hello_world_elf";

  std::ifstream test_elf_stream{test_elf_file};
  const std::string buffer{std::istreambuf_iterator<char>{test_elf_stream},
                           std::istreambuf_iterator<char>{}};
  ASSERT_NE(buffer.size(), 0);

  auto elf_file =
      ElfFile::CreateFromBuffer(test_elf_file, buffer.data(), buffer.size());
  ASSERT_NE(elf_file, nullptr);
} 
```

It would be great if you could have another quick look.